spiceypy/spiceypy.py

```diff
@@ -77,6 +78,34 @@
 _SPICE_EK_MAXQSEL = 100  # Twice the 50 in gcc-linux-64
 _SPICE_EK_EKRCEX_ROOM_DEFAULT = 100  # Enough?
 
+import threading
+
+_spicelock = threading.RLock()
+
+def spicelock_for_multithread(f):
+    """
+    Decorator for spiceypy to avoid concurrent calls to cspice lib.
+
+    :return:
+    """
+    @functools.wraps(f)
+    def lock(*args, **kwargs):
+        if config.enable_threading_lock:
+            with _spicelock:
```

Comment:
using [contextlib.suppress](https://docs.python.org/3/library/contextlib.html#contextlib.suppress) here instead could remove the if statement and the duplicated code below in the else clause.
I believe this would work:
```python
ctx = _spicelock if config.enable_threading_lock else contextlib.suppress()
with ctx: 
    try:
    ....
```

Reply:
Thanks, I am not familiar with context manager. It is a good point to increase the readability. Thanks

Reply:
I forgot to mention in the initial post to add a note reminder that once python 3.6 is deprecated the statement could switch to use 'nullcontext' instead which was introduced in python 3.7.  Maybe this post is a sufficient reminder to myself...